opentelemetry-tracing/opentelemetry-java-tracing/okhttp-and-jedis/src/main/java/com/splunk/tracing/otel/examples/javaagent/App.java

```diff
@@ -0,0 +1,97 @@
+package com.splunk.tracing.otel.examples.javaagent;
+
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.Span;
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import redis.clients.jedis.Jedis;
+
+/**
+ * This is a very simple demo app that shows the use of both the Java agent's
+ * auto-instrumentation combined with manual tracing.
+ * <p>
+ * The app accepts a single CLI argument that should be a URL.  The app will go
+ * out and fetch that URL with the OkHttp client (auto-instrumented) and then
+ * store it into a local Redis database using the Jedis client library (also
+ * auto-instrumented).  Both of these operations are performed within a root
+ * parent span called "fetch-and-set" that is manually created in the code
+ * below.
+ * <p>
+ */
+public class App {
+    private final OkHttpClient httpClient;
+    private final Jedis redisClient;
+    private final ExecutorService executor;
+
+    App() {
+        String redisHost = System.getenv("REDIS_HOSTNAME");
+        if (redisHost == null) {
+            redisHost = "localhost";
+        }
+        this.redisClient = new Jedis(redisHost);
+        this.httpClient = new OkHttpClient.Builder().build();
+        executor = Executors.newSingleThreadExecutor();
+    }
+
+    public static void main(String[] argv) {
+        if (argv.length == 0) {
+            System.out.println("Please specify a URL to fetch");
+            System.exit(1);
+        }
+
+        String url = argv[0];
+
+        App app = new App();
+        app.doFetchAndSet(url);
+    }
+
+    private void doFetchAndSet(String url) {
+        Request request = new Request.Builder()
+                .url(url)
+                .build();
+
+        // This span acts as a root span that envelops the sets of spans generated
+        // by both the HTTP call (using the OKHttp auto-instrumentation) and the
+        // Redis set action (using the redis-jedis auto-instrumentation).  If you
+        // did not have this root span that is manaully created, you would instead
+        // see two independent traces for the HTTP get and the Redis set.
+        // The Java agent sets the GlobalTracer instance before the application's main method is called.
+        Span span = OpenTelemetry.getTracer("sample").spanBuilder("fetch-and-set").startSpan();
```

Comment:
Store `OpenTelemetry.getTracer("sample")` to a local variable. It is bad example to call it twice in a row